packages/perf/src/index.ts

```diff
@@ -0,0 +1,172 @@
+import { logger } from '@libp2p/logger'
+import { anySignal } from 'any-signal'
+import { MAX_INBOUND_STREAMS, PROTOCOL_NAME, TIMEOUT, WRITE_BLOCK_SIZE } from './constants.js'
+import type { Connection } from '@libp2p/interface/src/connection/index.js'
+import type { Startable } from '@libp2p/interface/startable'
+import type { ConnectionManager } from '@libp2p/interface-internal/connection-manager'
+import type { IncomingStreamData, Registrar } from '@libp2p/interface-internal/registrar'
+import type { AbortOptions } from '@libp2p/interfaces'
+
+const log = logger('libp2p:perf')
+
+export const defaultInit: PerfServiceInit = {
+  protocolName: '/perf/1.0.0',
+  maxInboundStreams: 1 << 10,
+  maxOutboundStreams: 1 << 10,
+  timeout: 10000,
+  writeBlockSize: BigInt(64 << 10)
+}
+
+export interface PerfService {
+  perf: (connection: Connection, sendBytes: bigint, recvBytes: bigint, options?: AbortOptions) => Promise<void>
+  measureDownloadBandwidth: (connection: Connection, size: bigint) => Promise<number>
```

Comment:
Do we need these methods? afaict there helpers to calling perf with a 0. Would be nicer to have a smaller api.